### PR TITLE
Add command for converting various Byron keys to their Shelley counterpart

### PR DESCRIPTION
**Pull request by intricate**
Add support for converting:
 * payment keys
 * genesis keys
 * genesis delegate keys

Support either verification or signing keys.

Support current or legacy (ie SL) formats.

Also add special conversion for Byron genesis verification keys in base64,
as they appear in the Byron genesis file as `issuerPK`.

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Key.hs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+data ShelleyKeyCmdError
+  = ShelleyKeyCmdReadFileError !(FileError TextEnvelopeError)
+  | ShelleyKeyCmdWriteFileError !(FileError ())
+  | ShelleyKeyCmdByronKeyFailure !ByronKeyFailure
+  | ShelleyKeyCmdByronKeyParseError
+      !Text
+      -- ^ Text representation of the parse error. Unfortunately, the actual
+      -- error type isn't exported.
+  | ShelleyKeyCmdDeserialiseByronVerKeyError
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Key.hs:40]: I feel like it'd be nicer to instead have something like `ShelleyKeyCmdCborDecoderError !DecoderError`.

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Key.hs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+runConvertByronPaymentKey skeyPathOld (SigningKeyFile skeyPathNew) = do
+    sk <- firstExceptT ShelleyKeyCmdByronKeyFailure $
+            readEraSigningKey ByronEra skeyPathOld
+    firstExceptT ShelleyKeyCmdWriteFileError . newExceptT $
+      writeFileTextEnvelope skeyPathNew (Just skeyDesc) (ByronSigningKey sk)
+  where
+    skeyDesc = TextViewDescription "Payment Signing Key"
+
+runConvertByronGenesisVerificationKey
+  :: VerificationKeyFile -- ^ Input file: old format
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Key.hs:78]: Perhaps we should accept this either as a file or as a string CLI arg.

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 3]
```diff
@@ -265,6 +251,49 @@ pDelegationFee =
       <> Opt.help "The delegation fee in Lovelace."
       )
 
+pKeyCmd :: Parser KeyCmd
+pKeyCmd =
+  Opt.subparser $
+    mconcat
+      [ Opt.command "convert-byron-key"
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:258]:
```suggestion
      [ Opt.command "convert-byron-payment-signing-key"
```

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 4]
```diff
@@ -267,14 +269,30 @@ pKeyCmd =
...
     pKeyConvertITNKey =
       KeyConvertITNStakeKey
-        <$> pITNKeyFile
+        <$> pITNKeyFIle
         <*> pMaybeOutputFile
 
-    pITNKeyFile :: Parser ITNKeyFile
-    pITNKeyFile = pITNSigningKeyFile <|> pITNVerificationKeyFile
+    pITNKeyFIle :: Parser ITNKeyFile
+    pITNKeyFIle = pITNSigningKeyFile <|> pITNVerificationKeyFile
```
Comment (dcoutts) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs]: Accidental name change ("I" vs "i").

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Key.hs (hunk 2 above)
Comment (dcoutts) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Key.hs:78]: this is a rarely used command, we only need to support one

**Comment by dcoutts** [comment]
bors merge

**Comment by intricate** [comment]
I can't approve since I originally opened this PR, but LGTM :100:

**Pull request merged**